### PR TITLE
fix: secure sql queries generated by leaderboard stats

### PR DESCRIPTION
**Pull request by ghusse**
I switched back to the previous way of generating SQL queries in leaderboard-stat-getter, for several reasons:

1. The SQL query needed to be secured, the parameter `limit` was prone to SQL injection, and we cannot guarantee that other parameters won't introduce a SQL injection in the future
2. The generate SQL query was only compatible with Postgresql, because of the double quotes. MySQL uses back-ticks for securing column and table names
3. The generated SQL query was not correctly converting field names into column names

This PR correctly handles all these cases, and introduces new integration tests on this class, to be sure that everything is working as expected:
1. Tests are run on multiple databases
2. all table names contain spaces, in order to make sure that they are correctly escaped
3. all field names on which the aggregation is done contain spaces too, to be sure that everything is converted from field names to column names, and correctly escaped 

Linked to CU-d90u5g

## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Create automatic tests
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by arnaudbesnier** [comment]
Task linked: [CU-d90u5g High vulnerability - SQL injections](https://app.clickup.com/t/d90u5g)

**Review by VincentMolinie — APPROVED** [review]

File: src/services/leaderboard-stat-getter.js [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+import { InvalidParameterError } from './errors';
 
+function getAggregateField({
+  aggregateField, schemaRelationship, modelRelationship,
+}) {
+  // NOTICE: As MySQL cannot support COUNT(table_name.*) syntax, fieldName cannot be '*'.
+  const fieldName = aggregateField
+    || schemaRelationship.primaryKeys[0]
+    || schemaRelationship.fields[0].field;
+  return `${modelRelationship.name}.${Orm.getColumnName(schemaRelationship, fieldName)}`;
```
[the remark this answers is not in the document]
Comment (ghusse) [reply src/services/leaderboard-stat-getter.js:13]: I think I tested without this function, and it does not work.

File: src/services/leaderboard-stat-getter.js [hunk 2]
```diff
@@ -15,50 +35,52 @@ function LeaderboardStatGetter(model, modelRelationship, params, options) {
...
-  `;
+  const labelColumn = Orm.getColumnName(schema, labelField);
+  const groupBy = `${associationAs}.${labelColumn}`;
 
+  this.perform = async () => {
+    const records = await modelRelationship
+      .unscoped()
+      .findAll({
+        attributes: [
+          options.sequelize.col(groupBy),
```
[the remark this answers is not in the document]
Comment (ghusse) [reply src/services/leaderboard-stat-getter.js:60]: I will definitively try that

File: src/services/leaderboard-stat-getter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghusse) [reply src/services/leaderboard-stat-getter.js:13]: I tested again, and I did not find any solution that did not use this method to generate a valid column name

**Review by VincentMolinie — APPROVED** [review]

**Pull request merged**